src/lib.rs

```diff
@@ -36,6 +36,25 @@ pub enum CowArc<'a, T: ?Sized + 'static> {
     Owned(Arc<T>),
 }
 
+impl<'a: 'static, T: 'static> CowArc<'static, T> {
```

Comment:
Looks like `'a` is unused 
```suggestion
impl<T: 'static> CowArc<'static, T> {
```